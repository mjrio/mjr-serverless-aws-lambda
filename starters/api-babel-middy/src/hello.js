import middy from 'middy';
import {
  warmup,
  jsonBodyParser,
  cors,
  httpErrorHandler,
} from 'middy/middlewares';
import { json } from './utils/rest';

// handler
const hello = async (event, context) => {
  const message = 'Go Serverless! Your function executed successfully!';
  console.log(event, context);
  return json({ message });
};

const isWarmingUp = (event) => event.isWarmingUp === true;
const onWarmup = (event) => console.log('I am just warming up', event);

export default middy(hello)
  .use(warmup({ isWarmingUp, onWarmup }))
  .use(jsonBodyParser())
  .use(cors({ origin: '*', credentials: true }))
  .use(httpErrorHandler());
