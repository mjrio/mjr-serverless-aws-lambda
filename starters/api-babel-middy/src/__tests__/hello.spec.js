import { promisify } from 'util';
import hello from '../hello';

// const promisify = (handler) => (event, context) =>
//   new Promise((resolve, reject) => {
//     handler(event, context, (err, result) => {
//       if (err) {
//         return reject(err);
//       }
//       return resolve(result);
//     });
//   });

describe('hello', () => {
  it('executes as expected', async () => {
    const asyncHandler = promisify(hello);
    const result = await asyncHandler({ test: 1212 }, {});
    expect(result).toMatchSnapshot();
  });
});
