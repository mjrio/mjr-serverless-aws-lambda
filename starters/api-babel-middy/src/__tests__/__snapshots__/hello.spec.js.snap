// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello executes as expected 1`] = `
Object {
  "body": "{\\"message\\":\\"Go Serverless! Your function executed successfully!\\"}",
  "statusCode": 200,
}
`;
